src/wfs/src/test/java/org/geoserver/wfs/TransactionListenerTest.java

```diff
@@ -86,18 +86,30 @@ public void testInsert() throws Exception {
                 + "</cgf:lineStringProperty>"
                 + "<cgf:id>t0002</cgf:id>"
                 + "</cgf:Lines>"
+                + "<cgf:Points>"
+                + "<cgf:pointProperty>"
+                + "<gml:Point><gml:coordinates>15,15</gml:coordinates></gml:Point>"
+                + "</cgf:pointProperty>"
+                + "<cgf:id>t0005</cgf:id>"
+                + "</cgf:Points>"
                 + "</wfs:Insert>"
                 + "</wfs:Transaction>";
 
         postAsDOM("wfs", insert);
-        assertEquals(2, listener.events.size());
+        assertEquals(4, listener.events.size());
 
         TransactionEvent firstEvent = listener.events.get(0);
         assertTrue(firstEvent.getSource() instanceof InsertElementType);
         assertEquals(TransactionEventType.PRE_INSERT, firstEvent.getType());
         assertEquals(CiteTestData.LINES, firstEvent.getLayerName());
+
+        TransactionEvent thirdEvent = listener.events.get(2);
+        assertTrue(thirdEvent.getSource() instanceof InsertElementType);
+        assertEquals(TransactionEventType.PRE_INSERT, thirdEvent.getType());
+        assertEquals(CiteTestData.POINTS, thirdEvent.getLayerName());
+
         // one feature from the pre-insert hook, one from the post-insert hook
```

Comment:
While it's testing the insert order, the test has become confusing. It was meant to test pre/post insert events, now it's doing two things and some of the comments do not line up any longer.

Could you write a separate test instead, with a name identifying the intent of checking insertion order, and assertions in it that only check that?